src/utxo/repo.rs

```diff
@@ -445,4 +445,33 @@ impl UtxoRepo {
 
         Ok(usize::try_from(res.average).expect("Could convert to usize"))
     }
+
+    pub async fn delete_utxo(
+        &self,
+        tx: &mut Transaction<'_, Postgres>,
+        outpoint: bitcoin::OutPoint,
+        keychain_id: KeychainId,
+    ) -> Result<LedgerTransactionId, UtxoError> {
+        let row = sqlx::query!(
+            r#"DELETE FROM bria_utxos 
+            WHERE tx_id = $1 AND vout = $2 AND keychain_id = $3 AND income_settled_ledger_tx_id IS NULL
```

Comment:
follow order of `UNIQUE` index

Reply:
remove `income_settled_ledger_tx_id` from where clause and check in code (in order to return an error)